elastic/beats
# Allow credential_profile_name and shared_credential_file with role_arn

## What does this PR do?

This PR is to revisit the `role_arn` config parameter for AWS and add support for `role_arn` to be used with 
`credential_profile_name` and `shared_credential_file`.

## Why is it important?

Sometimes, the default credential profile might not be at the location that AWS SDK thinks is. So user should be able to specify that instead of only relying on the default location and default profile name.

## Checklist




## How to test this PR locally

1. Setup role ARN following the steps updated in the documentation.
2. Enable aws module and the config below should work
```
- module: aws
  period: 5m
  role_arn: arn:aws:iam::123456789012:role/example-role
  shared_credential_file: /Users/kaiyansheng/.aws/credentials_backup
  credential_profile_name: test
  metricsets:
    - ec2
```

## Related issues

- Relates https://github.com/elastic/integrations/issues/738
